src/Pass/ImgTagTransformPass.php

```diff
@@ -72,22 +72,33 @@ function pass()
                 continue;
             }
             $context_string = $this->getContextString($dom_el);
-            $new_dom_el = $this->cloneAndRenameDomElement($dom_el, 'amp-img');
-            $new_el = $el->prev();
-
-            $success = $this->setResponsiveImgHeightAndWidth($new_el);
-            // We were not able to get the image dimensions, abort conversion.
-            if (!$success) {
-                $this->addActionTaken(new ActionTakenLine('img', ActionTakenType::IMG_COULD_NOT_BE_CONVERTED, $lineno, $context_string));
-                // Abort the conversion and remove the new img tag
-                $new_el->remove();
-                continue;
+            if ($this->isPixel($el)) {
```

Comment:
Seems like github swallowed my other comment. Here is it again: can you please extract each if branch into a separate method. e.g.

```
if ($this->isPixel($el)) {
   convertAmpPixel(...)
} else {
  convertAmpImg(...)
} 
```
This makes it a bit easier to understand